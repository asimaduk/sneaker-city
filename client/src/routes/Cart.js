import React, { Component } from 'react';

import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Divider from '@material-ui/core/Divider';
import { Grid } from '@material-ui/core';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import { Link } from 'react-router-dom';

import '../App.css';

class Cart extends Component {
    constructor(){
        super();

        this.state = {
            items: []
        }
    }

    componentDidMount = (props) => {
        //calling a global function for any items in cart
        if(window.getCart){
            const cartItems = window.getCart();

            if(cartItems && cartItems.length > 0){
                let total = 0;

                document.title = `Cart (${cartItems.length}) | Sneaker City`;

                cartItems.forEach(item=> {
                    total += item.price;
                })

                this.setState({items: cartItems, total});
            }
            else {
                document.title = `Cart | Sneaker City`;
            }
        }
        else {
            document.title = `Cart | Sneaker City`;
        }
    }

    removeFromCart = (currentProduct) => {
        //filtering with product id and size
        const cartItems = this.state.items.filter(product => {
            if((product.id === currentProduct.id) && (product.size === currentProduct.size)){
                return false;
            }
            
            return true;
        });

        document.title = `Cart (${cartItems.length}) | Sneaker City`;
        this.setState({items: cartItems});
        
        if(window.removeFromCart){
            window.removeFromCart(currentProduct);
        }
    }

    render(){
        const { items, total } = this.state;

        return (
            <div className="Cart-Div">
                {items.length > 0 ?
                    <Grid container xs={12} sm={12} lg={6} spacing={24} style={{padding: 24, marginTop: 30}}>
                        <List aria-label="main mailbox folders" style={{width:'100%'}}>
                            {items.map((currentProduct, index) => (
                                <div style={{width:'100%'}}>
                                <ListItem className="" style={{width:'100%'}}>
                                    <img src={currentProduct.image} alt="alt" style={{width:'100px', marginRight:'20px'}}/>
                                    
                                    <div style={{width:'100%', flexDirection:'column', justifyContent:'space-between', minHeight:'110px',display:'flex'}}>
                                        <Grid container spacing={24} style={{padding: 0}}>
                                            <Grid item lg={11} >
                                                <div>
                                                    <h6 className="Cart-Item-Name">{currentProduct.name}</h6>
                                                    <h3 className="Cart-Item-Desc">{currentProduct.model}</h3>
                                                    <p className="Cart-Item-Size">Size: {currentProduct.size}</p>
                                                </div>
                                            </Grid>
                                            <Grid item lg={1} >
                                                <IconButton aria-label="close" onClick={()=> this.removeFromCart(currentProduct)}>
                                                    <CloseIcon />
                                                </IconButton>
                                            </Grid>
                                        </Grid>

                                        <Grid container spacing={24} style={{padding: 0}}>
                                            <Grid item lg={10} >
                                                <div>
                                                    x 1
                                                </div>
                                            </Grid>
                                            <Grid item lg={2} >
                                            <p style={{width:'100%', textAlign:'right', margin:0}}> $ {currentProduct.price}</p>
                                                {/* <div style={{display:'flex',alignItems:'center', flexDirection:'row' justifyContent:'right'}}>
                                                    
                                                </div> */}
                                            </Grid>
                                        </Grid>
                                    </div>
                                </ListItem>
                                <Divider />
                                </div>
                            ))}
                        </List>

                        <div style={{width:'100%',minHeight:'30px',display: 'flex', alignItems: 'center', flexDirection:'row', justifyContent:'space-between', paddingLeft:'16px',paddingRight:'13px'}}>
                            <p style={{display:'inline'}}>Total</p>
                            <p style={{display:'inline'}}>$ {total}</p>
                        </div>

                        <Link to={{pathname: `/checkout`, product: []}} style={{width:'100%',minHeight:'30px',display: 'flex', alignItems: 'center', flexDirection:'row', justifyContent:'center', paddingLeft:'16px',paddingRight:'13px', backgroundColor:'#ccc', textDecoration:'none', borderRadius:'5px', color:'#000', marginTop:'7px'}}>
                            <p style={{display:'inline'}}>PROCEED TO CHECKOUT</p>
                        </Link>
                    </Grid>
                    :
                    <h3>No item in cart</h3>
                }
            </div>
        )
    }
}

export default Cart;
