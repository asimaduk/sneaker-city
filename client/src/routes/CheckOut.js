import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Stepper from '@material-ui/core/Stepper';
import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import Button from '@material-ui/core/Button';
import Link from '@material-ui/core/Link';
import Typography from '@material-ui/core/Typography';

import AddressForm from '../components/AddressForm';
import PaymentForm from '../components/PaymentForm';
import ReviewOrder from '../components/ReviewOrder';

function Copyright() {
    return (
        <Typography variant="body2" color="textSecondary" align="center">
            {'Copyright Â© '}
            <Link color="inherit" href="/">
                Sneaker City
            </Link>{' '}
            {new Date().getFullYear()}
            {'.'}
        </Typography>
    )
}

const useStyles = makeStyles(theme => ({
    appBar: {
        position: 'relative',
    },
    layout: {
        width: 'auto',
        marginLeft: theme.spacing(2),
        marginRight: theme.spacing(2),
        [theme.breakpoints.up(600 + theme.spacing(2) * 2)]: {
            width: 600,
            marginLeft: 'auto',
            marginRight: 'auto',
        }
    },
    paper: {
        marginTop: theme.spacing(3),
        marginBottom: theme.spacing(3),
        padding: theme.spacing(2),
        [theme.breakpoints.up(600 + theme.spacing(3) * 2)]: {
            marginTop: theme.spacing(6),
            marginBottom: theme.spacing(6),
            padding: theme.spacing(3),
        }
    },
    stepper: {
        padding: theme.spacing(3, 0, 5),
    },
    buttons: {
        display: 'flex',
        justifyContent: 'flex-end',
    },
    button: {
        marginTop: theme.spacing(3),
        marginLeft: theme.spacing(1),
    }
}));
  
const steps = ['Shipping address', 'Payment details', 'Review your order'];
  
function getStepContent(step) {
    switch (step) {
        case 0:
            return <AddressForm />;
        case 1:
            return <PaymentForm />;
        case 2:
            return <ReviewOrder />;
        default:
            throw new Error('Unknown step');
    }
}

export default function Checkout() {
    const classes = useStyles();
    const [activeStep, setActiveStep] = React.useState(0);
  
    document.title = `Checkout | Sneaker City`;

    const handleNext = () => {
        if(activeStep === steps.length - 1){
            if(window.emptyCart){
                window.emptyCart();
            }
            
            setActiveStep(activeStep + 1);
        }
        else if(activeStep === 0){
            if(window.address){
                if(!window.address.firstname  || (window.address.firstname && !window.address.firstname.trim())){
                    alert('First name required.')
                }
                else if(!window.address.lastname || (window.address.lastname && !window.address.lastname.trim())){
                    alert('Last name required.')
                }
                else if(!window.address.address1 || (window.address.address1 && !window.address.address1.trim())){
                    alert('Address 1 required.')
                }
                else if(!window.address.city || (window.address.city && !window.address.city.trim())){
                    alert('City required.')
                }
                else if(!window.address.zip || (window.address.zip && !window.address.zip.trim())){
                    alert('Zip required.')
                }
                else if(!window.address.country || (window.address.country && !window.address.country.trim())){
                    alert('Country required.')
                }
                else {
                    setActiveStep(activeStep + 1);
                }
            }
            else {
                alert('Please provide address.')
            }
        }
        else if(activeStep === 1){
            if(window.paymentCard){
                if(!window.paymentCard.name  || (window.paymentCard.name && !window.paymentCard.name.trim())){
                    alert('Card name required.')
                }
                else if(!window.paymentCard.number || (window.paymentCard.number && !window.paymentCard.number.trim())){
                    alert('Card number required.')
                }
                else if(!window.paymentCard.expiryDate || (window.paymentCard.expiryDate && !window.paymentCard.expiryDate.trim())){
                    alert('Card expiry date required.')
                }
                else if(!window.paymentCard.cvv || (window.paymentCard.cvv && !window.paymentCard.cvv.trim())){
                    alert('Card CVV required.')
                }
                else {
                    setActiveStep(activeStep + 1);
                }
            }
            else {
                alert('Please provide card details.')
            }
        }
        else {
            setActiveStep(activeStep + 1);
        }
    };

    const handleBack = () => {
        setActiveStep(activeStep - 1);
    };

    return (
        <React.Fragment>
            <main className={classes.layout}>
                <Paper className={classes.paper}>
                    <Typography component="h1" variant="h4" align="center">
                        Checkout
                    </Typography>
                    
                    <Stepper activeStep={activeStep} className={classes.stepper}>
                        {steps.map(label => (
                            <Step key={label}>
                                <StepLabel>{label}</StepLabel>
                            </Step>
                        ))}
                    </Stepper>

                    <React.Fragment>
                        {activeStep === steps.length ? (
                            <React.Fragment>
                                <Typography variant="h5" gutterBottom>
                                    Thank you for your order.
                                </Typography>

                                <Typography variant="subtitle1">
                                    Your order number is #2001539. We have emailed your order confirmation, and will
                                    send you an update when your order has shipped.
                                </Typography>
                            </React.Fragment>
                        ) : (
                        <React.Fragment>
                            {getStepContent(activeStep)}
                            
                            <div className={classes.buttons}>
                                {activeStep !== 0 && (
                                    <Button onClick={handleBack} className={classes.button}>
                                        Back
                                    </Button>
                                )}

                                <Button
                                    variant="contained"
                                    color="primary"
                                    onClick={handleNext}
                                    className={classes.button}>
                                    {activeStep === steps.length - 1 ? 'Place order' : 'Next'}
                                </Button>
                            </div>
                        </React.Fragment>
                        )}
                    </React.Fragment>
                </Paper>
                
                <Copyright />
            </main>
        </React.Fragment>
    );
}