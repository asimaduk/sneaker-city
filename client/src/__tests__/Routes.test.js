import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';

import Cart from '../routes/Cart';
import CheckOut from '../routes/CheckOut';
import ProductDetails from '../routes/ProductDetails';

//tests routes for successful rendering

describe('Cart', () => {
    test('renders without crashing', () => {
        const component = create(<Cart />); 
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('CheckOut', () => {
    test('renders without crashing', () => {
        const component = create(<CheckOut />); 
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('ProductDetails', () => {
    const component = create(<ProductDetails />); 
    const tree = component.toJSON();

    test('renders without crashing', () => {
        expect(tree).toMatchSnapshot();
    });
});