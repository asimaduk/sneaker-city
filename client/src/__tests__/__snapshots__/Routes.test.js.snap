// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders without crashing 1`] = `
<div
  className="Cart-Div"
>
  <h3
    style={Object {}}
  >
    No item in cart
  </h3>
</div>
`;

exports[`CheckOut renders without crashing 1`] = `
<main
  className="makeStyles-layout-14"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-15 MuiPaper-rounded"
  >
    <h1
      className="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
    >
      Checkout
    </h1>
    <div
      className="MuiPaper-root MuiPaper-elevation0 MuiStepper-root MuiStepper-horizontal makeStyles-stepper-16"
    >
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                1
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
            >
              Shipping address
            </span>
          </span>
        </span>
      </div>
      <div
        className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                2
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Payment details
            </span>
          </span>
        </span>
      </div>
      <div
        className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
      >
        <span
          className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <div
        className="MuiStep-root MuiStep-horizontal"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
        >
          <span
            className="MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiStepIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                className="MuiStepIcon-text"
                textAnchor="middle"
                x="12"
                y="16"
              >
                3
              </text>
            </svg>
          </span>
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              Review your order
            </span>
          </span>
        </span>
      </div>
    </div>
    <div>
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
      >
        Shipping address
      </h6>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
              htmlFor="firstName"
            >
              First name
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="fname"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="firstName"
                name="firstName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
              htmlFor="lastName"
            >
              Last name
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="lname"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="lastName"
                name="lastName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
              htmlFor="address1"
            >
              Address line 1
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="billing address-line1"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="address1"
                name="address1"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="address2"
            >
              Address line 2
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="billing address-line2"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="address2"
                name="address2"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
              htmlFor="city"
            >
              City
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="billing address-level2"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="city"
                name="city"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="state"
            >
              State/Province/Region
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="state"
                name="state"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
              htmlFor="zip"
            >
              Zip / Postal code
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="billing postal-code"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="zip"
                name="zip"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
              htmlFor="country"
            >
              Country
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="billing country"
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="country"
                name="country"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <label
            className="MuiFormControlLabel-root"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-290 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  className="PrivateSwitchBase-input-293"
                  data-indeterminate={false}
                  name="saveAddress"
                  onChange={[Function]}
                  type="checkbox"
                  value="yes"
                />
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
            <span
              className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Use this address for payment details
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-buttons-17"
    >
      <button
        className="MuiButtonBase-root MuiButton-root makeStyles-button-18 MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Next
        </span>
      </button>
    </div>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
  >
    Copyright © 
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
      href="/"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      Sneaker City
    </a>
     
    2019
    .
  </p>
</main>
`;

exports[`ProductDetails renders without crashing 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 300,
      "justifyContent": "center",
    }
  }
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
